### DISCUSSION TITLE
/orm/migrations.html

### DISCUSSION THREAD

**Discussion by giscus** [goravel/goravel#324] [Announcements]
# /orm/migrations.html

[[toc]] 简介 当多人协作开发应用程序时，如果同步数据库结构没有一个统一的规范，以保证所有人的本地数据都是一致的，那将是灾难。数据库迁移就是为了解决这个问题，将数据库的结构进行版本控制，以保证所有开发人员的数据库结构的一致性。 生成迁移 使用 make:migration 命令来创建迁移： 该命令会在 database/migrations 目录...

https://www.goravel.dev/zh/orm/migrations.html

**Comment by agomezguru** [comment]
Hello everybody:

When get this error:
Migration rollback failed: Dirty database version 20240501230455. Fix and force version.

What do I need to do?
By now,  for me the only way is recover from the backup, but when you are developing is tedious

**Reply by hwbrzzl** [reply]
Yes, I agree, this behavior will be optimized in this PR: https://github.com/goravel/framework/pull/439

**Comment by wongvio** [comment]
快捷生成用啥：
sh-4.2# go run . artisan migrate create_users_table
sh-4.2# go run . artisan migrate:create_users_table
sh-4.2# go run . artisan create_users_table

都不对哎

**Reply by hwbrzzl** [reply]
https://www.goravel.dev/zh/orm/migrations.html#%E7%94%9F%E6%88%90%E8%BF%81%E7%A7%BB

```
go run . artisan make:migration create_users_table
```

**Comment by wongvio** [comment]
目前还没有删除字段，修改字段的功能吗

**Reply by hwbrzzl** [reply]
目前只支持原生 sql 语句进行修改，预计 1.15 支持链式。

**Reply by halfstring** [reply]
上面说的migrations支持链式，可有时间表么？在github discussions没看到呢~~~

**Reply by krishankumar01** [reply]
@halfstring There is an issue related to this feature: https://github.com/goravel/goravel/issues/280

**Comment by hulutech-web** [comment]
very good !!!

**Comment by SamuelBenedix** [comment]
How if i want use UUID for ID ?

**Reply by hwbrzzl** [reply]
Here: https://github.com/goravel/framework/blob/3cde68c2c6c3fee56816d4122c1149e309ff691c/support/database/database_test.go#L34
